src/main/java/com/cburch/logisim/std/memory/ShiftRegister.java

```diff
@@ -424,16 +438,15 @@ public void paintInstance(InstancePainter painter) {
   private void paintInstanceEvolution(InstancePainter painter) {
     // draw boundary, label
     painter.drawLabel();
-    int xpos = painter.getLocation().getX();
-    int ypos = painter.getLocation().getY();
-    BitWidth widObj = painter.getAttributeValue(StdAttr.WIDTH);
-    int wid = widObj.getWidth();
-    Integer lenObj = painter.getAttributeValue(ATTR_LENGTH);
-    int len = lenObj == null ? 8 : lenObj;
-    Boolean parallelObj = painter.getAttributeValue(ATTR_LOAD);
-    boolean Negedge = painter.getAttributeValue(StdAttr.EDGE_TRIGGER).equals(StdAttr.TRIG_FALLING);
-    DrawControl(painter, xpos, ypos, len, wid, parallelObj, Negedge);
-    ShiftRegisterData data = (ShiftRegisterData) painter.getData();
+    final var xpos = painter.getLocation().getX();
+    final var ypos = painter.getLocation().getY();
+    final var wid = painter.getAttributeValue(StdAttr.WIDTH).getWidth();
+    final var lenObj = painter.getAttributeValue(ATTR_LENGTH);
+    final var len = lenObj == null ? 8 : lenObj;
+    final var parallelObj = painter.getAttributeValue(ATTR_LOAD);
+    final var negedge = painter.getAttributeValue(StdAttr.EDGE_TRIGGER).equals(StdAttr.TRIG_FALLING);
```

Comment:
Unless I understand origins of this name wrong, negEdge please (with Edge capitalized).